dynomitemanager-core/src/main/java/com/netflix/dynomitemanager/storage/RedisStorageProxy.java

```diff
@@ -128,15 +127,24 @@ private void startPeerSync(String peer, int port) {
 
         localRedisConnect();
 
+        /*
+         * Iterate until we succeed the SLAVEOF command with some sleep
+         * time in between. We disconnect and reconnect if the jedis connection
+         * fails.
+         */
         while (!isDone) {
             try {
                 // only sync from one peer for now
                 isDone = (this.localJedis.slaveof(peer, port) != null);
```

Comment:
if this connect failed, localJedis is null

Reply:
yes, and we get an exception. So we the node remains unhealthy and does not join the ring.